src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs

```diff
@@ -1517,6 +1517,34 @@ public void Add(string fileName, string entryName)
 		}
 
 
+		/// <summary>
+		/// Add a file to the archive.
+		/// </summary>
+		/// <param name="fileName">The name of the file to add.</param>
+		/// <param name="entryName">The name to use for the <see cref="ZipEntry"/> on the Zip file created.</param>
+		/// <param name="entryDateTime">The DateTime to use for the <see cref="ZipEntry"/> on the Zip file created.</param>
+		/// <exception cref="ArgumentNullException">Argument supplied is null.</exception>
+		public void Add(string fileName, string entryName, DateTime entryDateTime)
+		{
+			if (fileName == null)
+			{
+				throw new ArgumentNullException(nameof(fileName));
+			}
+
+			if (entryName == null)
+			{
+				throw new ArgumentNullException(nameof(entryName));
+			}
+
+			CheckUpdating();
+			contentsEdited_ = true;
+
+			ZipEntry entry = EntryFactory.MakeFileEntry(entryName);
```

Comment:
`ZipEntryFactory.MakeFileEntry(string fileName)` is an overload of `MakeFileEntry(fileName, null, true)` which means that this will try to read the file `entryName` and create an entry from that file.

If you change this to `ZipEntry entry = EntryFactory.MakeFileEntry(fileName, entryName, true);` it will work as intended.

I am not too sure about how useful this addition is, though. Adding an entry with a custom `DateTime` can be done through `ZipFile.Add(ZipEntry)`.